features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -35,11 +39,19 @@ internal class Base64Serializer private constructor(
     private val base64LRUCache: Cache<Drawable, CacheData>,
     private val bitmapPool: BitmapPool?,
     private val logger: InternalLogger,
-    private val md5HashGenerator: MD5HashGenerator
+    private val md5HashGenerator: MD5HashGenerator,
+    private val recordedDataQueueHandler: DataQueueHandler,
+    private val applicationId: String
 ) {
     private var isBase64CacheRegisteredForCallbacks: Boolean = false
     private var isBitmapPoolRegisteredForCallbacks: Boolean = false
 
+    // resources previously sent in this session -
+    // optimization to avoid sending the same resource multiple times
+    // atm this set is unbounded but expected to use relatively little space (~80kb per 1k items)
+    private val resourcesSeen: MutableSet<String> =
```

Comment:
minor: after a bit of more thinking, let's emphasise that we store IDs here, and not some values, by renaming to `resourceIdsSeen`. wdyt?